helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/ClusterModelProvider.java

```diff
@@ -75,9 +154,24 @@ public static ClusterModel generateClusterModel(ResourceControllerDataProvider d
     // Check if the replicas need to be reassigned.
     Map<String, Set<AssignableReplica>> allocatedReplicas =
         new HashMap<>(); // <instanceName, replica set>
-    Set<AssignableReplica> toBeAssignedReplicas =
-        findToBeAssignedReplicas(replicaMap, clusterChanges, activeInstances,
-            dataProvider.getLiveInstances().keySet(), bestPossibleAssignment, allocatedReplicas);
+    Set<AssignableReplica> toBeAssignedReplicas;
+    switch (scopeType) {
+      case GLOBAL:
+        toBeAssignedReplicas = findToBeAssignedReplicasByClusterChanges(replicaMap, activeInstances,
+            dataProvider.getLiveInstances().keySet(), clusterChanges, bestPossibleAssignment,
+            allocatedReplicas);
+        break;
+      case PARTIAL:
+        // For partial rebalance, new partitions that are not in the Baseline won't be rebalanced.
+        // Filter to remove the replicas that are not in the Baseline from the replica map.
```

Comment:
"replicas that do not exist in the baseline given but exist in replicaMap. This is because such replicas are new additions that do not need to be rebalanced right away."